src/GitTfs/Core/TfsWorkspace.cs

```diff
@@ -177,7 +177,14 @@ private TfsPolicyOverrideInfo GetPolicyOverrides(CheckinOptions options, IChecki
 
         public string GetLocalPath(string path)
         {
-            return Path.Combine(_localDirectory, path);
+            var localPath = Path.Combine(_localDirectory, path);
+
+            if (System.IO.File.Exists(localPath))
+            {
+                return localPath;
+            }
+
+            return this._workspace.GetLocalItemForServerItem($"{Remote.TfsRepositoryPath}/{path}");
```

Comment:
I haven't thought much about this, so take my comments with a grain of salt.

With the above change, we now we have different behavior **if a file exists or not** at a specific path,
which makes me nervous. Should we not always return `this._workspace.GetLocalItemForServerItem($"{Remote.TfsRepositoryPath}/{path}")`